app/views/citizens/transactions/show.html.erb

```diff
@@ -8,29 +8,43 @@
 
     <%= form.hidden_field :transaction_type, value: @transaction_type.name %>
 
-    <table class="govuk-table">
+    <table class="govuk-table sortable">
       <%= content_tag(:caption, t('date.date_period', from: date_from, to: date_to), class: 'govuk-table__caption') if @bank_transactions.any? %>
       <thead class="govuk-table__head">
         <tr class="govuk-table__row">
-          <th class="govuk-table__header" scope="col"><%= t '.col_select_all' %></th>
-          <th class="govuk-table__header" scope="col"><%= t '.col_date' %></th>
-          <th class="govuk-table__header" scope="col"><%= t '.col_description' %></th>
-          <th class="govuk-table__header govuk-table__header--numeric" scope="col"><%= t '.col_amount' %></th>
+          <th class="govuk-table__header" scope="col">
+            <a
+              class="govuk-body select-clear-all"
+              href=""
+              data-copy-select="<%= t '.col_select_all' %>"
+              data-copy-clear="<%= t '.col_clear_all' %>"></a>
+          </th>
+          <th class="govuk-table__header sort header-sort-asc" scope="col"><%= t('.col_date') %></th>
+          <th class="govuk-table__header sort" scope="col"><%= t('.col_description') %></th>
+          <th class="govuk-table__header govuk-table__header--numeric sort" scope="col"><%= t('.col_amount') %></th>
         </tr>
       </thead>
       <tbody class="govuk-table__body">
         <%= form.collection_check_boxes :transaction_ids, @bank_transactions, :id, :description do |builder| %>
           <tr class="govuk-table__row">
             <th class="govuk-table__cell">
-              <div class="govuk-checkboxes__item">
-                <% checked = builder.object.transaction_type_id == @transaction_type.id %>
-                <%= builder.check_box(class: 'govuk-checkboxes__input', checked: checked) %>
-                <label class="govuk-label govuk-checkboxes__label"></label>
-              </div>
+              <% if builder.object.transaction_type_id == nil || builder.object.transaction_type_id == @transaction_type.id %>
```

Comment:
Would this work here:
```erb
<% if builder.object.transaction_type_id ||= @transaction_type.id %>
```

Reply:
I think that's different.
your code assigns the transaction_type we want to select to all transactions that are not assigned to any transaction type yet

Reply:
Sorry - you're right.